### DISCUSSION TITLE
How to connect to a NATS server configured with self-signed certificate, from natscli?

### DISCUSSION THREAD

**Discussion by aDisplayName** [nats-io/nats.net#597] [Q&A] [answered]
If a nats server is configured using self-signed TLS certificate (.crt, .key), how to connect to this server using nats cli?

I would assume that the client side needs to provide the self-signed CA file, from which those self-signed TLS certificates were issued, to the nats cli using `--tls` option, is that correct?

**Comment by mtmk** [comment, marked as answer]
that's right. I think you need to use `--tlsca=FILE  TLS certificate authority chain ($NATS_CA)` option. see also `nats --help` cc @ripienaar

**Reply by aDisplayName** [reply]
Thank you. Much appreciated
